compiler/rustc_hir_analysis/src/hir_ty_lowering/errors.rs

```diff
@@ -1496,6 +1497,32 @@ fn generics_args_err_extend<'a>(
             ];
             err.multipart_suggestion_verbose(msg, suggestion, Applicability::MaybeIncorrect);
         }
+        GenericsArgsErrExtend::DefVariant(segments) => {
+            let args: Vec<Span> = segments
+                .iter()
+                .filter_map(|segment| match segment.res {
+                    Res::Def(
+                        DefKind::Ctor(CtorOf::Variant, _) | DefKind::Variant | DefKind::Enum,
+                        _,
+                    ) => segment.args().span_ext().map(|s| s.with_lo(segment.ident.span.hi())),
+                    _ => None,
+                })
+                .collect();
+            if args.len() > 1
+                && let Some(span) = args.into_iter().last()
+            {
```

Comment:
```suggestion
            if let [.., arg] = *args {
```

Reply:
`..` can match 0 elements so `&[1]` will match a pattern `[.., 1]`, you would need `if let [_, .., span]` for this to do what you want i think. This seems somewhat weird to write whereas existing logic seems very straightforward to me